packages/esm-patient-forms-app/src/htmlformentry/html-form-entry-wrapper.component.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { useEffect } from 'react';
+import styles from './html-form-entry-wrapper.scss';
+
+interface HtmlFormEntryWrapperProps {
+  closeWorkspace: () => void;
+  src: string;
+}
+
+const HtmlFormEntryWrapper: React.FC<HtmlFormEntryWrapperProps> = ({ closeWorkspace, src }) => {
+  useEffect(() => {
+    const iframe = document.querySelector('iframe');
+    iframe.addEventListener('load', () => {
+      // hide the header and breadcrumbs
+      const dashboard = iframe.contentDocument;
+      dashboard.querySelector('header')?.remove();
+      dashboard.querySelector('.patient-header')?.remove();
+      dashboard.querySelector('#breadcrumbs')?.remove();
+      iframe.style.display = 'block';
+    });
+
+    // register a listener to close the workspace when HFE sends a "close-workspace" message
+    window.addEventListener('message', (event) => {
```

Comment:
Calling `addEventListener()` without `removeEventListener()` leaves the event listener around. This is usually not to be a deal in a multi-page application, since each page load will remove the listeners, but in a SPA it's essential to clean things up. I'd probably use a separate `useEffect()` hook for this like this:

```ts
useEffect(() => {
	const callback = (event) => event?.data === 'close-workspace' && closeWorkspace();
  window.addEventListener('message', callback);
  // functions returned from `useEffect()` get called when the component containing the hook is unmounted
  return () => window.removeEventListener('message', callback);
}, [closeWorkspace])
```